File: src/main/java/org/mifos/fineract/services/CashierJournalsApi.java
package org.mifos.fineract.services;

import retrofit2.http.GET;
import retrofit2.http.Headers;
import rx.Observable;

public interface CashierJournalsApi {
    /**
     * @param officeId  (optional)
     * @param tellerId  (optional)
     * @param cashierId (optional)
     * @param dateRange (optional)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("cashiersjournal")
    Observable<String> getJournalData(
            @retrofit2.http.Query("officeId") Long officeId, @retrofit2.http.Query("tellerId") Long tellerId, @retrofit2.http.Query("cashierId") Long cashierId, @retrofit2.http.Query("dateRange") String dateRange
    );

}

File: src/main/kotlin/org/openapitools/client/apis/CashierJournalsApi.kt
package org.openapitools.client.apis

import retrofit2.http.GET
import retrofit2.http.Query


interface CashierJournalsApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param officeId  (optional)
     * @param tellerId  (optional)
     * @param cashierId  (optional)
     * @param dateRange  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/cashiersjournal")
    suspend fun getJournalData1(
        @Query("officeId") officeId: Long? = null,
        @Query("tellerId") tellerId: Long? = null,
        @Query("cashierId") cashierId: Long? = null,
        @Query("dateRange") dateRange: String? = null
    ): String

}
